cmd/xl-v1.go

```diff
@@ -181,9 +181,9 @@ func (xl xlObjects) StorageInfo(ctx context.Context, local bool) StorageInfo {
 	disks := xl.getDisks()
 	if local {
 		var localDisks []StorageAPI
-		for i, disk := range disks {
+		for _, disk := range disks {
 			if disk != nil {
-				if xl.endpoints[i].IsLocal && disk.Hostname() == "" {
```

Comment:
This also means you can remove xl.endpoints @vadmeste NOTE: all command line args should be same 

So not sure if this the right fix still @vadmeste

Reply:
> So not sure if this the right fix still @vadmeste

yes you are right, xl.endpoints is also passed later in the same function, so something is wrong, we do not want to reorder endpoints list right ? I see it called from many places in xl sets.

Reply:
If the underlying disks get reordered it's okay, but not endpoints and MinIO shouldn't start with different ordered endpoints.

Reply:
> If the underlying disks get reordered it's okay, but not endpoints and MinIO shouldn't start with different ordered endpoints.

I've changed the way to fix this. It is more simple now